pkg/test/framework/components/galley/snapshot.go

```diff
@@ -48,13 +48,31 @@ type SnapshotValidatorFunc func(actuals []*SnapshotObject) error
 // is found in the snapshot.
 func NewSingleObjectSnapshotValidator(ns string, fn func(ns string, actual *SnapshotObject) error) SnapshotValidatorFunc {
 	return func(actuals []*SnapshotObject) error {
-		if len(actuals) != 1 {
+		filteredActuals := actualsByNamespace(ns, actuals)
+		if len(filteredActuals) != 1 {
 			return fmt.Errorf("expected 1 resource, found %d", len(actuals))
 		}
 		return fn(ns, actuals[0])
 	}
 }
 
+func actualsByNamespace(ns string, actuals []*SnapshotObject) (result []*SnapshotObject) {
+	for _, a := range actuals {
+		if ns == getNamespace(a.Metadata.Name) {
+			result = append(result, a)
+		}
+	}
+	return result
+}
+
+func getNamespace(metadataName string) string {
```

Comment:
@ozevren I know there are a couple of utilities (e.g. FullName) inside Galley that do this. However, the MCP metadata just has name as a string. Is there a preferred utility to parse it, or should all the call sites roll their own?

Reply:
I think this is fairly simple and mechanical, and defined as part of the MCP documentation, so it should be ok to have external implementations.
`resource.Name` in `galley/pkg/config/resource` and related methods are the library methods that I'd recommend for this.